lib/transformers/excel.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 Mia s.r.l.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict'
+
+const { Transform } = require('stream')
+const XLSXTransformStream = require('xlsx-write-stream')
+
+module.exports = () => ({
+  stringifier: () => {
+    let headerProcessed = false
+    const dataTransformer = new Transform({
+      transform(chunk, _, callback) {
+        if (!headerProcessed) {
+          headerProcessed = true
+          const columns = Object.keys(chunk)
```

Comment:
This operation can be performed safely because we are assuming that all the records respect the same schema (due to the collection being managed by the CRUD Service). Is this correct?

Otherwise we may not be able to extract all the headers processing only the first record.

Reply:
>This operation can be performed safely because we are assuming that all the records respect the same schema (due to the collection being managed by the CRUD Service). Is this correct?

yes this is the assumption

Reply:
a smarter way could be implemetned by taking the schema and use ti to create columns but then we would have to make sure the order for each document is respected

Reply:
OK, sounds fair